src/main/scala/is/hail/expr/Type.scala

```diff
@@ -545,7 +545,7 @@ final case class TAggregableVariable(elementType: Type, st: Box[SymbolTable]) ex
     throw new RuntimeException("TAggregableVariable is not realizable")
 }
 
-final case class TVariable(name: String, var t: Type = null) extends Type {
+final case class TVariable(name: String, var t: Type = null, predicate: Type => Boolean = (t) => true) extends Type {
```

Comment:
prefer `_ => true` since `t` isn't used and there is another `t` hovering nearby.